### PR TITLE
fix: CType does not check credential ctype in presentation

### PR DESCRIPTION
**Pull request by Dudleyneedham**
## fixes [#3391](https://github.com/KILTprotocol/ticket/issues/3391)

Updating the verify credential to check against the credentials ctypes id against the given ctype

## How to test:
Please provide a brief step-by-step instruction.
If necessary provide information about dependencies (specific configuration, branches, database dumps, etc.)

- Step 1
- Step 2
- etc.

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [x] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [ ] I have documented the changes (where applicable)
    * Either PR or Ticket to update [the Docs](https://github.com/KILTprotocol/docs)
    * Link the PR/Ticket here

### CODE REVIEW

File: packages/legacy-credentials/src/Credential.ts [hunk 1]
```diff
@@ -298,8 +298,14 @@ export function verifyWellFormed(
 ): void {
   verifyDataStructure(credential)
   verifyDataIntegrity(credential)
-
+  const credentialCTypeId = CType.hashToId(credential.claim.cTypeHash)
```
Comment (rflechtner) [inline packages/legacy-credentials/src/Credential.ts:301]: can you move that into the if clause?

File: packages/legacy-credentials/src/Credential.ts [hunk 2]
```diff
@@ -298,8 +298,14 @@ export function verifyWellFormed(
...
   if (ctype) {
+    if (credentialCTypeId !== ctype.$id) {
+      throw new SDKErrors.CTypeIdMismatchError(
+        ctype.$id,
+        credential.claim.cTypeHash
```
Comment (rflechtner) [inline packages/legacy-credentials/src/Credential.ts:306]: and maybe that can be the `credentialCTypeId` as well then?

**Review by rflechtner — APPROVED** [review]
LGTM but the variable assignment can be moved into the if clause

File: packages/legacy-credentials/src/Credential.ts (hunk 1 above)
Comment (Dudleyneedham) [reply packages/legacy-credentials/src/Credential.ts]: https://github.com/KILTprotocol/sdk-js/pull/867/commits/56bc559147a83ccb16418dd20bb143af32726500

File: packages/legacy-credentials/src/Credential.ts (hunk 2 above)
Comment (Dudleyneedham) [reply packages/legacy-credentials/src/Credential.ts]: https://github.com/KILTprotocol/sdk-js/pull/867/commits/56bc559147a83ccb16418dd20bb143af32726500

**Pull request merged**